uncertainty_baselines/models/resnet50_radial.py

```diff
@@ -0,0 +1,286 @@
+  epsilon=1e-5,  # using epsilon and momentum defaults from Torch
+  momentum=0.9)
+Conv2DFlipout = functools.partial(  # pylint: disable=invalid-name
+  ed.layers.Conv2DFlipout, use_bias=False)
+
+
+def get_radial_kernel_regularizer(
+    tied_mean_prior: bool, radial_prior: bool):
+  """
+  Determine the regularizer function based on provided prior settings.
+
+  Args:
+    tied_mean_prior: bool, if True, fix the mean of the prior to that of
+      the variational posterior, which causes the KL to only penalize the
+      weight posterior's standard deviation, and not its mean.
+    radial_prior: bool, if True, use a Radial prior instead of a Normal.
+
+  :return: regularizer class
+  """
+  if tied_mean_prior:
+    if radial_prior:
+      kernel_regularizer = RadialKLDivergenceWithTiedMean
+    else:
+      kernel_regularizer = ed.regularizers.NormalKLDivergenceWithTiedMean
+  else:
+    if radial_prior:
+      kernel_regularizer = RadialKLDivergence
+    else:
+      kernel_regularizer = ed.regularizers.NormalKLDivergence
+
+  # Can optionally set the mean of the untied priors -
+  # we default to a standard Normal/Radial
+  return kernel_regularizer
+
+
+def bottleneck_block(inputs, filters, stage, block, strides, prior_stddev,
+                     dataset_size, stddev_init, tied_mean_prior=True,
+                     radial_prior=False):
+  """Residual block with 1x1 -> 3x3 -> 1x1 convs in main path.
+
+  Note that strides appear in the second conv (3x3) rather than the first (1x1).
+  This is also known as "ResNet v1.5" as it differs from He et al. (2015)
+  (http://torch.ch/blog/2016/02/04/resnets.html).
+
+  Args:
+    inputs: tf.Tensor.
+    filters: list of integers, the filters of 3 conv layer at main path
+    stage: integer, current stage label, used for generating layer names
+    block: 'a','b'..., current block label, used for generating layer names
+    strides: Strides for the second conv layer in the block.
+    prior_stddev: Fixed standard deviation for weight prior.
+    dataset_size: Dataset size to properly scale the KL.
+    stddev_init: float to initialize variational posterior stddev parameters.
+    tied_mean_prior: bool, if True, fix the mean of the prior to that of the
+      variational posterior, which causes the KL to only penalize the weight
+      posterior's standard deviation, and not its mean.
+    radial_prior: bool, if True, use a Radial prior instead of a Normal.
+
+  Returns:
+    tf.Tensor.
+  """
+  filters1, filters2, filters3 = filters
+  conv_name_base = 'res' + str(stage) + block + '_branch'
+  bn_name_base = 'bn' + str(stage) + block + '_branch'
+  kernel_regularizer = get_radial_kernel_regularizer(
+    tied_mean_prior=tied_mean_prior, radial_prior=radial_prior)
+
+  x = Conv2DFlipout(
+    filters1,
+    kernel_size=1,
+    kernel_initializer=TrainableHeRadial(
+      stddev_initializer=tf.keras.initializers.TruncatedNormal(
+        mean=np.log(np.expm1(stddev_init)), stddev=0.1)),
```

Comment:
Should 0.1 be hard coded here?

Reply:
Probably ought to be a hyper, good point!